evm_loader/rpc-client/src/http.rs

```diff
@@ -0,0 +1,157 @@
+            client: HttpClientBuilder::default().build(config.url)?,
+        })
+    }
+}
+
+pub struct NeonRpcHttpClientBuilder {}
+
+impl NeonRpcHttpClientBuilder {
+    pub fn new() -> NeonRpcHttpClientBuilder {
+        NeonRpcHttpClientBuilder {}
+    }
+
+    pub async fn build(&self, url: impl Into<String>) -> NeonRpcClientResult<NeonRpcHttpClient> {
+        let config = NeonRpcClientConfig::new(url);
+        NeonRpcHttpClient::new(config).await
+    }
+}
+
+impl Default for NeonRpcHttpClientBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[async_trait(?Send)]
+impl NeonRpcClient for NeonRpcHttpClient {
+    async fn cancel_trx(&self, params: CancelTrxRequest) -> NeonRpcClientResult<CancelTrxReturn> {
+        self.request(LibMethods::CancelTrx, params).await
+    }
+
+    async fn collect_treasury(&self) -> NeonRpcClientResult<CollectTreasuryReturn> {
+        self.request_without_params(LibMethods::CollectTreasury)
+            .await
+    }
+
+    async fn create_ether_account(
+        &self,
+        params: CreateEtherAccountRequest,
+    ) -> NeonRpcClientResult<CreateEtherAccountReturn> {
+        self.request(LibMethods::CreateEtherAccount, params).await
+    }
+
+    async fn deposit(&self, params: DepositRequest) -> NeonRpcClientResult<DepositReturn> {
+        self.request(LibMethods::Deposit, params).await
+    }
+
+    async fn emulate(
+        &self,
+        params: EmulateRequestModel,
+    ) -> NeonRpcClientResult<EmulationResultWithAccounts> {
+        self.request(LibMethods::Emulate, params).await
+    }
+
+    async fn get_ether_account_data(
+        &self,
+        params: GetEtherRequest,
+    ) -> NeonRpcClientResult<GetEtherAccountDataReturn> {
+        self.request(LibMethods::GetEtherAccountData, params).await
+    }
+
+    async fn get_neon_elf(
+        &self,
+        params: GetNeonElfRequest,
+    ) -> NeonRpcClientResult<GetNeonElfReturn> {
+        self.request(LibMethods::GetNeonElf, params).await
+    }
+
+    async fn get_storage_at(
+        &self,
+        params: GetStorageAtRequest,
+    ) -> NeonRpcClientResult<GetStorageAtReturn> {
+        self.request(LibMethods::GetStorageAt, params).await
+    }
+
+    async fn init_environment(
+        &self,
+        params: InitEnvironmentRequest,
+    ) -> NeonRpcClientResult<InitEnvironmentReturn> {
+        self.request(LibMethods::InitEnvironment, params).await
+    }
+
+    async fn trace(&self, params: TraceRequestModel) -> NeonRpcClientResult<serde_json::Value> {
+        self.request(LibMethods::Trace, params).await
+    }
+}
+
+impl NeonRpcHttpClient {
+    async fn request<R, P>(&self, method: LibMethods, params: P) -> NeonRpcClientResult<R>
+    where
+        P: jsonrpsee_core::Serialize, // + jsonrpsee_core::traits::ToRpcParams + Send,
+        R: jsonrpsee_core::DeserializeOwned,
+    {
+        let client = self.client.clone();
+        Ok(client.request(method.into(), rpc_params![params]).await?)
```

Comment:
There is no need for `.clone()` call. So it can be simplified to:
```Rust
        Ok(self
            .client
            .request(method.into(), rpc_params![params])
            .await?)
```

Reply:
fixed